src/views/Login/Keystore.js

```diff
@@ -69,83 +63,65 @@ class Keystore extends React.Component {
-        }),
-    });
-
-    const decryptMessage =
-      decryptionProblem === false ? (
-        <div />
-      ) : (
-        <Warning className={'mt-8'}>
-          <H3 style={{ marginTop: 0, paddingTop: 0 }}>
-            {"Couldn't decrypt wallet."}
-          </H3>
-          {'You may have entered an incorrect password.'}
-        </Warning>
-      );
-
-    return (
-      <Row>
-        <Col className={'measure-md'}>
-          <H1 className={'mb-4'}>{'Upload Your Keystore File'}</H1>
-          <InputCaption>
-            {`Please upload your Ethereum keystore file.  If your keystore
-             file is encrypted with a password, you'll also need to enter
-             that below.`}
-          </InputCaption>
-
-          <UploadButton
-            className={`btn ${uploadButtonClass} mt-10`}
-            onChange={this.handleKeystoreUpload}>
-            <div className={'flex-center-all fs-4 h-11 pointer'}>
-              {'Upload Keystore file'}
-            </div>
-          </UploadButton>
-
-          <Input
-            className="pt-8 mt-8"
-            prop-size="md"
-            prop-format="innerLabel"
-            type="password"
-            name="password"
-            onChange={this.handlePasswordInput}
-            value={password}
-            autocomplete="off"
-            autoFocus>
-            <InnerLabel>{'Password'}</InnerLabel>
-          </Input>
-
-          <Button
-            className={` mt-10`}
-            prop-size={'wide lg'}
-            disabled={Nothing.hasInstance(keystore)}
-            onClick={this.constructWallet}>
-            {'Decrypt'}
-          </Button>
-
-          {decryptMessage}
-
-          <Button
-            className={'mt-10'}
-            prop-size={'wide lg'}
-            disabled={Nothing.hasInstance(wallet)}
-            onClick={this.props.loginCompleted}>
-            {'Continue →'}
-          </Button>
-        </Col>
-      </Row>
+  const uploadButtonClass = keystore.matchWith({
+    Nothing: _ => 'bg-blue white',
+    Just: ks =>
+      ks.value.matchWith({
+        Ok: _ => 'bg-green white',
+        Error: _ => 'bg-yellow black',
+      }),
+  });
+
+  const decryptMessage =
+    decryptionProblem === false ? (
+      <div />
+    ) : (
+      <Warning className={'mt-8'}>
+        <H3 style={{ marginTop: 0, paddingTop: 0 }}>
+          {"Couldn't decrypt wallet."}
+        </H3>
+        {'You may have entered an incorrect password.'}
+      </Warning>
     );
-  }
-}
 
-export default compose(withWallet)(Keystore);
+  return (
+    <View>
+      <InputCaption>
+        {`Please upload your Ethereum keystore file.  If your keystore
+         file is encrypted with a password, you'll also need to enter
+         that below.`}
+      </InputCaption>
+
+      <UploadButton
+        className={`${uploadButtonClass} mt3`}
+        onChange={handleKeystoreUpload}>
+        <div className={'flex-center-all fs-4 h-11 pointer'}>
+          {'Upload Keystore file'}
```

Comment:
You don't need the `{''}` syntax here—`<div>Upload Keystore file</div>` is semantically identical. Likewise across all of these files. the lone `{' '}` is only required when a formatted newline should be a space instead of being automatically collapsed by react.